examples/vision/siamese.py

```diff
@@ -0,0 +1,290 @@
+
+"""
+# Siamese network with a contrastive loss
+
+Author: [Mehdi](https://github.com/s-mrb)<br>
+Date created: 2021/05/06<br>
+Last modified: 2020/05/06<br>
+Description: Similarity learning using siamese network with contrastive loss
+"""
+
+"""
+## Introduction
+
+[Siamese Network](https://en.wikipedia.org/wiki/Siamese_neural_network)
+is any Neural Network which share weights between two or more sister networks,
+each producing embedding vector of its respective input and these embeddings
+are then passed through some 
+[distance heuristic](https://developers.google.com/machine-learning/clustering/similarity/measuring-similarity)
+to find the distance between them. This distance is later used to increase the
+contrast between embeddings of inputs of different classes and decrease it with
+that of similar class by employing some loss function, with the main objective
+of contrasting [vector spaces](https://en.wikipedia.org/wiki/Vector_space)
+from which these sample inputs were taken.
+
```

Comment:
@8bitmp3 hope this intro of Siamese works.

Reply:
Do what you think is best, I'm an outside contributor here 😄